api/upload.py

```diff
@@ -204,14 +255,42 @@ def engine(self):
             self.abort(400, 'container id is required')
         else:
             cid = bson.ObjectId(cid)
+
         context = {
             'job_id': self.get_param('job'),
             'job_ticket_id': self.get_param('job_ticket'),
         }
-        if level == 'analysis':
-            return process_upload(self.request, Strategy.analysis_job, origin=self.origin, container_type=level, id_=cid, context=context)
+
+        # Request for upload ticket
+        if self.get_param('ticket') == '':
+            return self._create_ticket()
+
+        # Check ticket id and skip permissions check if it clears
+        ticket_id = self.get_param('ticket')
```

Comment:
Having URL params `'ticket'` _and_ `'job_ticket'` on the same endpoint seems very confusing. I would suggest renaming this one to `'upload_ticket'` unless there's a problem with that I don't see yet.